### ISSUE TITLE
missing KLD

### ISSUE THREAD

**Issue by matthewvowels1** [zacheberhart/Maximum-Mean-Discrepancy-Variational-Autoencoder-MMD-VAE#1]
From what I can tell in the original paper, for MNIST the loss is as follows:

BCE - (1- alpha)KLD - (alpha + lambda - 1)MMD

where they set alpha = 0 and lambda = 1000.

i.e. the KL is not removed. I have a feeling that this model is incomplete and should incorporate either a sample-based estimation of KL, or an analytical estimation (using the learned mu and logvar params). Something like this, perhaps?

```
def loss_function(recon_x, x, z, mu, logvar ):
	true_samples = torch.randn(batch_size, latent_dim, requires_grad = False).to(device)
	mmd = lambda_ * compute_mmd(true_samples, z)
	KLD = -0.5 * torch.sum(1 + logvar - mu.pow(2) - logvar.exp())
	BCE = F.binary_cross_entropy(recon_x, x, reduction='sum')
	total_info = BCE + KLD + mmd
	return BCE, KLD, mmd, total_info
```
